src/cdk-experimental/dialog/dialog-container.ts

```diff
@@ -241,34 +245,72 @@ export class CdkDialogContainer extends BasePortalOutlet implements OnDestroy {
   }
 
   /**
-   * Autofocus the first tabbable element inside of the dialog, if there is not a tabbable element,
-   * focus the dialog instead.
+   * Focuses the provided element. If the element is not focusable, it will add a tabIndex
+   * attribute to forcefully focus it. The attribute is removed after focus is moved.
+   * @param element The element to focus.
    */
-  private _autoFocusFirstTabbableElement() {
+  private _forceFocus(element: HTMLElement, options?: FocusOptions) {
```

Comment:
Does it make sense to move this out into an a11y helper? We have an identical function in our g3 app.

Reply:
My understanding is that these methods will exist in cdk/dialog once that's created in a future PR

Reply:
Yeah, I told Amy to do it this way since cdk/dialog is the _correct_ place for this logic. It just has the unfortunately property of not existing yet.